src/SLCore/State/ActiveConfigScopeTracker.cs

```diff
@@ -34,9 +34,14 @@ namespace SonarLint.VisualStudio.SLCore.State;
 public interface IActiveConfigScopeTracker : IDisposable
 {
     ConfigurationScope Current { get; }
+
     void SetCurrentConfigScope(string id, string connectionId = null, string sonarProjectKey = null);
+
     void Reset();
+
     void RemoveCurrentConfigScope();
+
+    void UpdateRootOnCurrentConfigScope(string root);
```

Comment:
Not thread safe. You need to pass the configScopeId in case it has changed. Also it would be useful to make it return bool in that case.